lib/components/viewers/pattern-row.tsx

```diff
@@ -68,113 +72,63 @@ class PatternRow extends Component<Props, State> {
     }
 
     const routeName = route.shortName ? route.shortName : route.longName
+    const routeColor = getRouteColorBasedOnSettings(route.operator, route)
 
     return (
-      <div className="route-row" role="table">
+      <div className="route-row" role="group">
         {/* header row */}
-        <div className="header" role="row">
+        <div
+          className="header"
+          style={{
+            backgroundColor: routeColor,
+            color: getMostReadableTextColor(routeColor, route?.textColor)
+          }}
+        >
           {/* route name */}
           <div className="route-name">
-            <strong>
-              <RouteRenderer leg={generateFakeLegForRouteRenderer(route)} />
+            <strong
+              style={
+                routeName && routeName?.length >= 4 ? { fontSize: '3vb' } : {}
```

Comment:
I don't see that applied to the `RAPID` bus (try with http://localhost:9966/#/stop/2:841)

Reply:
Is this going to accommodate long long `routeShortName` that some agencies on the West Coast use?

Reply:
Yes if you have a look at septa which has long `routeShortName`s on it's rail lines they fit nicely.

Reply:
This item still needs to be resolved.